packages/orbit-components/src/hooks/useTransition/index.js

```diff
@@ -0,0 +1,83 @@
+// @flow
+import * as React from "react";
+
+import typeof UseTransitionType from ".";
+
+const useTransition: UseTransitionType = ({ show, appear = false }) => {
+  // if appear is true, we want to start from unmounted state, so that we can transition in
+  const [mounted, setMounted] = React.useState(!appear && show);
+  const [enter, setEnter] = React.useState(!appear && show);
+  const [done, setDone] = React.useState(!appear || !show);
+  const ref = React.useRef(null);
+  const firstRender = React.useRef(true);
+
+  // const [, updateState] = React.useState()
+  // const forceUpdate = React.useCallback(() => updateState({}), [])
+
+  // if (module.hot) {
+  //   module.hot.dispose(() => {
+  //     firstRender.current = true
+  //     setMounted(!appear && show)
+  //     setEnter(!appear && show)
+  //     setDone(!appear || !show)
+  //     forceUpdate()
+  //   })
+  // }
```

Comment:
I would like this to stay here for now because it's a way to keep this hook working through hot reloads, but I'm not sure how to include it so it doesn't end up in the build.